[wentingchua] iP

wentingchua opened the pull request:
### Dusty

> The best task management tool

Dusty is your personal task manager that helps you keep track of the things you need to do. It's

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use
- **FREE**
Here is how you can get started with Dusty:

1. Download it from [here](https://github.com/wentingchua/ip/releases/download/A-Jar/dusty.jar).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you!  😉

##### Sample commands

- [ ] `todo work on assignment`  creates a todo task
- [ ] `deadline essay /by 07/02/2025 2359` creates a deadline with stated date and time (in dd/mm/yyyy hh/mm format)
- [ ] `event meeting /from 08/02/2024 1000 /to 08/02/2024 1200` creates an event with stated date and time in (dd/mm/yyyy hh/mm format)

---

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

wentingchua committed:
Renamed 'Duke' to 'Dusty'

wentingchua committed:
Change greeting statement and added exit statement

wentingchua committed:
Enabled Dusty to echo command entered by user

wentingchua committed:
Enabled Dusty to exist when the user types the command bye

wentingchua committed:
Enabled Dusty to store text entered by user

wentingchua committed:
Enabled Dusty to display list

wentingchua committed:
Enabled Dusty to show tasks with empty checkboxes

wentingchua committed:
Enabled Dusty to mark tasks as completed

wentingchua committed:
Enabled Dusty to unmark

wentingchua committed:
Enabled Dusty to classify types of tasks

wentingchua committed:
enabled dusty to include due date for deadlines

wentingchua committed:
enabled dusty to identify deadline and event time from input

wentingchua committed:
handled errors for invalid format of output

wentingchua committed:
Enabled dusty to remove task according to input

wentingchua committed:
performed testings using automated text ui testing

wentingchua committed:
enabled saving of tasks into local storage

wentingchua committed:
enabled loading of tasks

wentingchua committed:
reformatted text for TaskStorage.java

wentingchua committed:
Merge branch 'branch-Level-7'

wentingchua committed:
enabled time formatter for deadline type tasks

wentingchua committed:
enabled time formatter for Event type tasks

wentingchua committed:
added tasks.txt to .gitignore

wentingchua committed:
Merge branch 'branch-Level-8'

wentingchua committed:
added files to use more OOP

wentingchua committed:
created packages and divided classes into them

wentingchua committed:
Merge remote-tracking branch 'origin/add-gradle-support'

wentingchua committed:
Fixed some bugs in relation to the date time formatter

wentingchua committed:
Merge branch 'branch-a-gradle'

wentingchua committed:
included JUnit test cases

wentingchua committed:
Edited build.gradle file to enable building of JAR file

wentingchua committed:
Added Javadocs to more frequently used classes and methods

wentingchua committed:
Made changes to improve style and readability

wentingchua committed:
Included additional case in Parser to handle find keyword. Amended UI accordingly.

wentingchua committed:
Merge branch 'branch-A-JavaDoc'

wentingchua committed:
Resolved merge conflicts

wentingchua committed:
Removed unnecessary comment for readability

wentingchua committed:
Merge branch 'branch-A-CodingStandard'

wentingchua committed:
Merge branch 'branch-Level-9'. Resolved merge conflicts.

JuhromeAlexander commented on src/main/java/backend/Parser.java:
```diff
@@ -0,0 +1,169 @@
+package backend;
+
+import store.*;
+import ui.*;
```
Perhaps the imports could be listed explicitly in order to not violate the coding standard.

JuhromeAlexander commented on src/main/java/backend/Parser.java:
```diff
@@ -0,0 +1,169 @@
+
+        try {
+            switch (command) {
+            case "list":
```
I like how you ensured that your case clauses for the switch statement has no indentations!

JuhromeAlexander commented on src/main/java/store/Storage.java:
```diff
@@ -0,0 +1,91 @@
+        if (!file.exists()) {
+            System.out.println("File does not exist. Creating it...");
+            if (file.getParentFile() != null && !file.getParentFile().exists()) {
+                boolean dirsCreated = file.getParentFile().mkdirs();
```
Perhaps the boolean variable could be renamed to something more indicative of the boolean variable? Maybe something like isDirectoryCreated would enhance readability of the code

JuhromeAlexander commented on src/main/java/store/Storage.java:
```diff
@@ -0,0 +1,91 @@
+                    Task task = Parser.parseTaskFromFile(line);
+                    if (task != null) {
+                        tasks.add(task);
+                    } else {
+                        System.out.println("Parser returned null for line: " + line);
+                    }
+                } catch (Exception e) {
+                    System.out.println("Error parsing line: " + line);
+                    e.printStackTrace();
```
I like how you handled the error of having a malformed line in the saved tasks file! 👍

JuhromeAlexander reviewed (commented):
LGTM! Just a few nits to fix.

wentingchua committed:
Improve import statements in Parser.java

wentingchua committed:
Improve boolean name in Storage.java

wentingchua committed:
Configure Checkstyle and make changes to adhere to Checkstyle

wentingchua committed:
Merge branch 'branch-A-CheckStyle'

Kkenjji commented on src/main/java/backend/Parser.java:
```diff
@@ -0,0 +1,174 @@
+        String command = parts[0];
+        String details = parts.length > 1 ? parts[1] : "";
+
+        try {
+            switch (command) {
+            case "list":
```
The `switch` cases handle different user commands, some have conditional blocks while others don't. 

Perhaps consider extracting each command out as separate classes? E.g. `AddCommand`, `ListCommand`

Kkenjji commented on src/main/java/backend/Parser.java:
```diff
@@ -0,0 +1,174 @@
+        String type = parts[0];
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+        Task task = null;
+
+        try {
+            switch (type) {
+            case "T":
+                task = new Task(description);
+                break;
+            case "D":
+                LocalDateTime by = DateTimeParser.parseDateTime(parts[3]);
+                task = new Deadline(description, by);
+                break;
+            case "E":
+                LocalDateTime from = DateTimeParser.parseDateTime(parts[3]);
+                LocalDateTime to = DateTimeParser.parseDateTime(parts[4]);
+                task = new Event(description, from, to);
+                break;
+            default:
+                break;
+            }
```
I like how OOP principle (polymorphism) is applied here!

Kkenjji commented on src/main/java/store/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+package store;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import backend.DateTimeParser;
```
Import statements are ordered correctly, nice!

Kkenjji commented on src/main/java/store/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+
+    @Override
+    public String toSaveFormat() {
+        return "D | " + getStatusIcon() + " | " + description + " | "
+                + by.format(DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm"));
```
Correct indentation after line break! 

Line break happens before an operator, which aligns with the coding convention!

Kkenjji commented on src/test/java/ui/DustyTest.java:
```diff
@@ -0,0 +1,99 @@
+        String actualOutput = outputStream.toString().trim();
+
+        // 1. Print Character Codes (CRUCIAL)
+        System.out.println("Expected (Character Codes):");
+        for (int i = 0; i < expectedOutput.length(); i++) {
+            System.out.print((int) expectedOutput.charAt(i) + " ");
+        }
+        System.out.println();
+
+        System.out.println("Actual (Character Codes):");
+        for (int i = 0; i < actualOutput.length(); i++) {
+            System.out.print((int) actualOutput.charAt(i) + " ");
+        }
+        System.out.println();
```
Would it be better to a a newline character `\n` at the end of the previous line instead of doing `System.out.println();`?

Kkenjji reviewed (commented):
LGTM. Just a few nits to consider.

wentingchua committed:
Integrate javafx into ip

wentingchua committed:
Align CLI commands to front end

wentingchua committed:
Merge branch 'branch-Level-10'

wentingchua committed:
Update DustyTest, Ui, and Checkstyle Configuration

DustyTest.java was incompatible with the new Main method. Ui methods
returned strings without printing them. Checkstyle configuration file
was not in the expected location, causing Gradle errors.

Update tests to align with the new application entry point. Print
responses to match with tests. Gradle’s Checkstyle check was failing due
to a missing or misplaced config file.

Refactor DustyTest.java to match the new Main method structure. Add
System.out.println() before every return statement in Ui. Move
checkstyle.xml to the correct location and updated build.gradle
accordingly.

Ensuring DustyTest aligns with the updated Main method prevents test
failures. Printing messages before returning them allows better
visibility while preserving method output. Relocating checkstyle.xml
ensures Gradle can find the file.

wentingchua committed:
Create different Java classes for the three of the commands

The method in the Parser.java class is becoming a long method,
especially if there is going to be more commands in the future.

Separate Java Classes created per command.

This ensures that each Command class executes their own unique set of
actions to complete that command within their own class.

wentingchua committed:
Refactor parser to delegate command execution to individual classes

Currently, Parser.java handles all command logic directly, making it
lengthy and difficult to maintain. Each command is processed within a
large switch statement, leading to long method and violation of SLAP.

This needs to change because having all command logic in one place
reduces modularity and makes extending or modifying command behavior
cumbersome.

Extract command execution logic into separate classes which extends from
Command.java. Modify Parser.java to match these changes.

This is done to improve code quality by following SLAP.

wentingchua committed:
Add assertion statements to Dusty.java, Main.java and MainWindow.java

wentingchua committed:
Merge pull request #1 from wentingchua/branch-A-Assertions

Add assertion statements to Dusty.java, Main.java and MainWindow.java

wentingchua committed:
Merge pull request #2 from wentingchua/branch-A-CodeQuality

Refactor parser to delegate command execution to individual classes to follow SLAP

wentingchua committed:
Merge branch 'master' of https://github.com/wentingchua/ip

wentingchua committed:
Add tag support to Todo, Event and Deadline to implement findtag functionality

wentingchua committed:
Merge branch 'Branch-C-Tagging'

wentingchua committed:
Change MarkCommand.java and UnmarkCommand.java to perform better
error handling for empty inputs

wentingchua committed:
Change DeleteCommand.java to handle invalid inputs better

wentingchua committed:
Change DeadlineCommand.java to handle invalid inputs better

wentingchua committed:
Change EventCommand.java to perform better error handling

wentingchua committed:
Add tagsToString() method in Task.java to convert tags to string
method for printing

wentingchua committed:
Enable tags to be printed on Deadline tasks

wentingchua committed:
Edit files to adhere to coding standards

wentingchua committed:
Added title into MainWindow.fxml

wentingchua committed:
Add product image and edit README

wentingchua committed:
Edit test cases for DustyTest.java

wentingchua committed:
Edit README to make user guide more specific

wentingchua committed:
Edit README to make it more readable

wentingchua committed:
Debug issues with Parser.java

wentingchua committed:
Fix naming error in MainWindow.java

wentingchua committed:
Debug error handling for DeleteCommand.java

wentingchua committed:
Edit README.md to update features

wentingchua committed:
Edit README.md to specify time format
